src/main/java/seedu/address/model/DateTime.java

```diff
@@ -0,0 +1,61 @@
+package seedu.address.model;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+import java.time.DateTimeException;
+import java.time.format.DateTimeFormatter;
+import java.util.Date;
+
+/**
+ * Represents a date or time (optional) within PalPay.
+ * Guarantees:  immutable; is valid as declared in {@link #isValidDate(String)}
+ */
+public class DateTime {
+
+    public static final String MESSAGE_CONSTRAINTS = "DateTime objects must adhere to the format: DDMMYYYY [HH:mm]";
+
+    public final String value;
+
+    public static final String VALIDATION_REGEX_DATE = "(0?[1-9]|[12][0-9]|3[01])(0?[1-9]|1[0-2])\\d{4}";
+
+    public static final String VALIDATION_REGEX_DATE_TIME = "(0?[1-9]|[12][0-9]|3[01])(0?[1-9]|1[0-2])\\d{4} (00|[0-9]|1[0-9]|2[0-3]):([0-9]|[0-5][0-9])";
```

Comment:
This should be the correct one. You can check again.
`(0?[1-9]|[12][0-9]|3[01])(0?[1-9]|1[0-2])(20[0-1][0-9]) (00|[01][0-9]|2[0-3]):(0[0-9]|[0-5][0-9])`

Reply:
As above